### PR TITLE
Use rcedit from PATH if not specified in Editor Settings

### PR DESCRIPTION
**Pull request by Calinou**
On non-Windows platforms, this will attempt to use a launcher script located in the PATH that will start the actual rcedit executable using WINE.

This makes it easier for power users to set up rcedit in an "universal" way that also works with Godot. This also makes it possible to use rcedit without creating a bespoke configuration file in CI environments.

Here's an example launcher script that can be dropped into `/usr/bin/rcedit` or another location in the PATH:

```bash
#!/bin/sh

DISPLAY=:0 WINEDEBUG=-all wine /usr/local/share/rcedit.exe "$@"
```

As far as I know, a launcher script is necessary because Windows executables are not directly callable on Linux/macOS (unless you use binfmt_misc, but that's more involved and only available on Linux).

### CODE REVIEW

File: platform/windows/export/export.cpp [hunk 1]
```diff
@@ -50,21 +50,24 @@ Error EditorExportPlatformWindows::export_project(const Ref<EditorExportPreset>
...
+	if (rcedit_path != String() && !FileAccess::exists(rcedit_path)) {
 		ERR_PRINTS("Could not find rcedit executable at " + rcedit_path + ", aborting.");
 		return ERR_FILE_NOT_FOUND;
 	}
 
+	if (rcedit_path == String()) {
+		// Try to run rcedit from PATH
+		// On non-Windows platforms, this will try to call a launcher script
+		// that uses WINE to run the actual rcedit executable
+		rcedit_path = "rcedit";
```
Comment (akien-mga) [inline platform/windows/export/export.cpp]: This is not really needed here, you can just check `if (rcedit_path == String)` directly before the `execute` call to do the branching.

**Comment by akien-mga** [comment]
Seems fine to me to look in the PATH, but there's one big issue here. With this PR, it's now impossible to export a Windows game without setting up rcedit yet (and thus WINE too on macOS and Linux), while previously it was optional (`return OK` if `rcedit_path == String()`).

**Comment by Calinou** [comment]
Exporting seems to work fine if rcedit is not present in the PATH and the rcedit path is empty in the Editor Settings, but this will be printed in the console:

```
**ERROR** OS_Unix::execute - Could not create child process while executing: rcedit
```

Should we leave it as-is, or is there a way to silence it?

**Comment by akien-mga** [comment]
I don't like the idea of automatically attempting to call a rcedit and having an error in the vast majority of cases where it won't be available.

**Comment by akien-mga** [comment]
Moving to 3.2 milestone as we're about to enter the beta stage and *release freeze* for Godot 3.1. Only bug fixes against issues of the 3.1 milestone (or enhancements to core features of the 3.1 roadmap) will be considered until 3.1-stable is released.

**Comment by akien-mga** [comment]
As the above issue still stands, I'm not sure this is worth merging as is. It's quite unlikely to have rcedit in the PATH unless you read Godot's documentation about it and know that you need it, and at this point you can just as well specify its path once in the Editor Settings.

**Pull request closed without merging**